### PR TITLE
Execute run-parts on /etc/subiquity/postinst.d before executing any other late_commands

### PR DESCRIPTION
**Pull request by ogayot**
This PR extends the early / late / error controllers so that they can declare builtin commands.
Builtin commands are executed before other commands declared in the relevant autoinstall section.

In the late controller, we now declare a builtin command that runs `run-parts /etc/subiquity/postinst.d`. The desktop installer can use this and place executable scripts under `/etc/subiquity/postinst.d` to copy network-manager connections and do other things.

### CODE REVIEW

File: subiquity/server/controllers/cmdlist.py [hunk 1]
```diff
@@ -25,6 +27,22 @@
 from subiquity.server.controller import NonInteractiveController
 
 
+@attr.s(auto_attribs=True)
+class Command:
+    args: Union[str, Sequence[str]]
+    check: bool
+
+    def desc(self) -> str:
```
Comment (dbungert) [inline subiquity/server/controllers/cmdlist.py:35]: I'm fine with this, just curious - was the built-in representation not great?

**Review by dbungert — APPROVED** [review]

File: subiquity/server/controllers/cmdlist.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/server/controllers/cmdlist.py:35]: I think it's just that this way of formatting is what is associated with the "start run" event. It's visible to the user iirc, so probably should stay friendly (I wonder if we should run it though shlex.quote though?)

**Review by mwhudson — APPROVED** [review]
Seems fine -- I have the same question as Dan.

File: subiquity/server/controllers/cmdlist.py (hunk 1 above)
Comment (ogayot) [reply subiquity/server/controllers/cmdlist.py:35]: So, the installation screen displays the commands that are run. When commands are specified as an array of arguments e.g., `["ls", "-l", "/"]`, they are transformed into a string to be more human readable.

I did not change the existing behavior here but moved the implementation in the Command object so we don't need that many branches in the the `run` function.

```diff
-            if isinstance(cmd, str):
-                desc = cmd
-            else:
-                desc = ' '.join(cmd)
+            desc = cmd.desc()
```

I will add some docstrings to make this clearer.

Comment (ogayot) [reply subiquity/server/controllers/cmdlist.py:35]: > (I wonder if we should run it though shlex.quote though?)

sure, let's `shlex.join` this. That will allow users/us to copy paste without losing information.

**Comment by ogayot** [comment]
Added a `try ... except AttributeError` block in the initializer of LateController.
`system_setup` imports this controller as well and does not have a postinst script mechanism.

**Pull request merged**